.github/workflows/test_on_push.yml

```diff
@@ -64,51 +61,35 @@ jobs:
     - name: Install standard python dependencies
       run: |
         python -m pip install --upgrade pip wheel setuptools
-        pip install -e .
+        python -m pip install tox
 
-    - name: Install SUNDIALS and SuiteSparse
+    - name: Install SuiteSparse and Sundials
       if: matrix.os != 'windows-latest'
-      run: |
-        pip install wget
-        git clone https://github.com/pybind/pybind11.git
-        python scripts/setup_KLU_module_build.py
-        export SUNDIALS_INST=$HOME/.local
-        pip install scikits.odes
-        pip install -e .
-
-    - name: Run unit tests Windows
-      if: matrix.os == 'windows-latest'
-      run: |
-        python run-tests.py --unit --folder all
+      run: tox -e pybamm-requires
 
-    - name: Run unit tests
+    - name: Run unit tests on GNU/Linux and MacOS
       if: matrix.os != 'windows-latest'
+      run: tox -e unit
+
+    - name: Run unit tests on Windows
+      if: matrix.os == 'windows-latest'
       run: |
-        export LD_LIBRARY_PATH=$HOME/.local/lib:scikits.odes/sundials5/lib:$LD_LIBRARY_PATH
+        pip install -e .
```

Comment:
so does tox not work on windows?

Reply:
yes it should, but configuring tox to work on both *nix and windows isnt straightforward (we don't want to install sundials and/or scikits.odes on windows). Initially I struggled to write a tox.ini that would deal with all the cases without being very messy and intricate. Eventually I gave up on Windows.

I tried again today and I thing I've got something that works. I had to change the structure of the tox.ini a lot though.

Bottom line is:

GNU/Linux & Mac
```bash
tox -e dev
source .tox/dev/bin/actvate
```

Windows:
```bash
python -m tox -e windows-dev
.tox\dev\Scripts\activate.bat
```